examples/cuda/02-broadcast.jl

```diff
@@ -0,0 +1,17 @@
+import MPI
+MPI.Init()
+comm = MPI.COMM_WORLD
+using CuArrays
+using CuArrays.CUDAdrv
+using CUDAnative
+CUDAnative.device!(MPI.Comm_rank(comm) % length(devices()))
```

Comment:
I believe the code below is a slightly better way to ensure the correct allocation of device id to rank on a machine.

```julia
lcomm = MPI.Comm_split_type(comm, MPI.MPI_COMM_TYPE_SHARED,  MPI.Comm_rank(comm))
CUDAnative.device!(MPI.Comm_rank(lrank))
```

Reply:
Thanks! I had no experience with Comm_split_type before.

Reply:
Thanks! I had no experience with GPUs over multiple machines.